pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/GreatestTransformFunction.java

```diff
@@ -0,0 +1,117 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.operator.transform.function;
+
+import org.apache.pinot.common.function.TransformFunctionType;
+import org.apache.pinot.core.operator.blocks.ProjectionBlock;
+
+
+public class GreatestTransformFunction extends SelectTupleElementTransformFunction {
+
+  public GreatestTransformFunction() {
+    super(TransformFunctionType.GREATEST.getName());
+  }
+
+  @Override
+  public int[] transformToIntValuesSV(ProjectionBlock projectionBlock) {
+    int numDocs = projectionBlock.getNumDocs();
+    if (_intValuesSV == null || _intValuesSV.length < numDocs) {
+      _intValuesSV = new int[numDocs];
+    }
+    int[] values = _arguments.get(0).transformToIntValuesSV(projectionBlock);
+    System.arraycopy(values, 0, _intValuesSV, 0, numDocs);
+    for (int i = 1; i < _arguments.size(); i++) {
+      values = _arguments.get(i).transformToIntValuesSV(projectionBlock);
+      for (int j = 0; j < numDocs & j < values.length; j++) {
```

Comment:
No need (and should not) to check the boundary of values, same for other places
```suggestion
      for (int j = 0; j < numDocs; j++) {
```

Reply:
Actually I did this for a reason - the invariant `numDocs < values.length` is not maintained anywhere and enforcing it allows bounds checks to be removed.